Devil Antag Round 2.

lordpidey opened the pull request:
Hello, this adds a new sideantag to SS13:  Soul merchants.
These devils go around, trying to buy the souls of the crew, and grow in power as they do so.

Please note that these are adminfuckery only.
(Admins, note this is done via a player's traitor panel on a current crew member)
:cl:
rscadd: Infernal devils have been seen onboard our spacestations, offering great boons in exchange for souls.
rscadd: Employees are reminded that their souls already belong to nanotrasen.  If you have sold your soul in error, a lawyer or head of personnel can help return your soul to Nanotrasen by hitting you with your employment contract.
rscadd: Nanotrasen headquarters will be bluespacing employment contracts into the Lawyer's office filing cabinet when a new arrival reaches the station.  It is recommended that the lawyer create copies of some of these for safe keeping.
rscadd: Due to the recent infernal incursions, the station library has been equipped with a Codex Gigas to help research infernal weaknesses.  Please note that reading this book may have unintended side effects.  Also note, you must spell the devil's name exactly, as there are countless demons with similar names.
rscadd: When a devil dies, if the proper banishment ritual is not performed on it's remains, the devil will revive itself at the cost of some of it's power.  The banishment ritual is described in the Codex Gigas.
rscadd: When a demon gains enough souls, It's form will mutate to a more demonic looking form.  The Arch-demon form is known to be on par with an ascended shadowling in power.
/:cl:

[x] create contracts
├ [X] create filing cabinet with employment contracts
├ [X] allow copier to copy employment contracts
├ [X] Add boons to willingly signed contracts
└ [X] Lawyer/HoP bashing someone with employment contract can return soul.

 [X] Add demon antag.
├ [X] Add demon objectives.
├ [X]  Add demon name generator.
├ [X] Add demon bans.  (A list of things the demon cannot/must do, similar to silicon laws)
│ ├ [X] Make demon's bane functional
│ ├ [X] Make ability to research demon's ban.
│ └ [X] Give librarian ability to research the demon's weaknesses.
│     └ [X]  Researching demons can have detrimental effects.
├ [X] Add demon spells/powers
├ [X] Sprites?
├ [X] Add admintools for adding/removing demons.
├ [X] Demons increase in power as they gain more souls
├ [X] Final form, Archdemon, is an instant shuttle call (akin to ascended shadowling)
└ [X] Demons lose power if they die, but resurrect.
    └ [X] Add rituals to prevent demonic resurrection.

 [X] Add Sintouched antags.
└ [X] Intentional exposure to demonic information causes sintouched.

Misc stuff:
[X] Add toy demonologist to arcade rewards.

Suggestions I've heard so far that I plan on implementing.
 [X] Move demonomicon from ground.
 [X] Alter the "magic" contract.
 [X] Make the employment cabinet look more important.
 [ ] Law diploma orderable from cargo, makes someone able to buy back souls (with the HoP's level of success, not the lawyer's)
 [ ] Spare demonomicons can be bought from cargo.
 [ ] Employment contract cabinet can be bought from cargo.

Criticism from the test on late 3/30-early 3/31
 [X] Replace infinite slaughter demons with imps.
 [X] Make imp simple mob.  Noticably weaker than slaughter demons.
 [X] Pitch fork needs ground icon.
 [X] Demonic resurrection didn't work, but it also didn't give a fail message.

See #15118 for the old PR

lordpidey committed:
Re-making the branch and PR of demon.  See #15118 for the old PR

lordpidey committed:
Adds toy demonologist to arcade machine reward list.

Supermichael777 commented:
:+1: Make sure at least one of the demon sprites has a pinstripe suit and tie.

Ultimate-Chimera commented:
:+1:

lordpidey commented:
I'm not sure if the soulless catching fire in the chapel is a good idea.  Selling your soul does NOT make you an antag.  However, having the sintouched catch fire in the chapel sounds like a cool idea.

AlexanderUlanH commented:
I like the sintouched objectives that are minor inconveniences, but it would be nice if some of the other objectives were more along those lines. I could see people reading the demonomicon intentionally to have an excuse to "break" the engine, punch people in maintenance, stun/cuff/strip people when they don't hand over their toolbelt and gloves.

Some ideas:
For greed; "Everything you see is a treasure - carry as much as possible."
Envy: "Why should you be stuck with your rank? Show everyone you can do other jobs too, and don't let anyone stop you, least of all because you have no training."
Pride: "You don't need anyone's help. So don't ask for it, and turn it away when you can."
Wrath: "What have your coworkers ever done for you? Don't offer to help them in any matter, and refuse if asked."

PKPenguin321 commented:
Like before, :+1:

lordpidey commented:
@AlexanderUlanH, The envy and wrath ones sound wonderful.  I'll use those.

KazeEspada commented on code/game/objects/items/weapons/twohanded.dm:
```diff
@@ -442,3 +442,47 @@
+/obj/item/weapon/twohanded/pitchfork
+	icon_state = "pitchfork0"
+	name = "pitchfork"
+	desc = "A red pitchfork, it looks like the work of the devil."
```
Make it a child of pitchfork, so that we can have normal metal pitchfork for ~~angry mobs~~ botanists.

KazeEspada commented on code/modules/paperwork/contract.dm:
```diff
@@ -0,0 +1,227 @@
+
+/obj/item/weapon/paper/contract/infernal/suicide_act(mob/user)
+	if(signed && (user == target.current))
+		user.say("OH GREAT INFERNO!  I DEMAND YOU COLLECT YOUR BOUNTY IMMEDIATELY!")
```
forcesay()

KazeEspada commented on code/modules/paperwork/contract.dm:
```diff
@@ -0,0 +1,227 @@
+	signed = 1
+	burn_timer += 10
+	update_text(user)
+	switch(contractType)
```
why a switch instead of a proc and subtypes?

lordpidey replied:
/obj/item/weapon/paper/contract/infernal/power  seems like an awfully long typestring, but okay, I'll change that.

KazeEspada commented:
Theres spaghetti everywhere. With Whiteness being the worst.

lordpidey committed:
Implemented two suggestions: Pitchfork now has a demonic subtype, to potentially give botanists a normal pitchfork.

Suiciding with signed contract now uses forcesay instead of say.

Robustin commented:
Since the summary doesn't really explain anything I'll ask:

1) What does selling a soul entail? Why would a crew member want to sell their soul? Why WOULDN'T they want to sell their soul? How would crew identify who had sold their soul? Are the soul-sellers antags? Can they attack the demon? 

2) How does the demonicon work, i.e. research? If you just have to type a name in, that doesn't seem hard at all. 

3) If the demonicon can give antag status to people, whats preventing the mass-antag meme from emerging where someone spots a demon, shouts about it, and suddenly the entire crew is poking around the demonicon until they get their antag status and go fuck with people. 

It sounds really interesting and I like that it gives roles to lesser positions, but generally modes that give crew lots of "voluntary" choices results in a complete shitshow. Also modes that involve conversion/snowballing/etc. tend to not go well with traitor at all.

lordpidey commented:
Since the summary doesn't really explain anything I'll ask:

> 1) What does selling a soul entail? 

The demon has a spell, summon contract.  You select a nearby person, and assuming the person isn't dead, you then select a contract type.  The contract appears in that person's hand, and using a pen on the contract signs it.   If the person IS dead, it will summon a contract of revival in the demon's hand, which when pressed to a corpse gives the ghost an option to sign the contract and come back.

> Why would a crew member want to sell their soul? 

Hulk, all access, unlimited minerals, magic spells, being revived, or antaghud + xray.  I'm open to suggestions for boons.

> Why WOULDN'T they want to sell their soul? 

Well for one, you are unrevivable after selling your soul.  Two: if the demon gets too many souls, it can ascend, and that's ALMOST as bad as an ascended shadowling.  Also RP reasons.

> How would crew identify who had sold their soul? 

You can't on a living person (though they could simply tell people), it's obvious on corpses with a suicide victim like message.

>Are the soul-sellers antags? Can they attack the demon?

No, and yes.  A demon selling someone hulk power, and then getting his face caved in is a real threat.

> 2) How does the demonicon work, i.e. research? If you just have to type a name in, that doesn't seem hard at all.

Yes, that's simply how it works, BUT you need the "demonic name" of the demon, not the name of his human shell.

> 3) If the demonicon can give antag status to people, whats preventing the mass-antag meme from emerging where someone spots a demon, shouts about it, and suddenly the entire crew is poking around the demonicon until they get their antag status and go fuck with people.

Time will.  for most crewmembers, it will take two and a half minutes straight on average to get antag from the demonomicon.  (This needs fixing in code right now, I'm figuring out the best way to do this atm)  Also, only one person can use the demonomicon at a time.

In order from fastest to slowest, here is the average  time for crewmembers to become sintouched.
Clown: 1.333 lookups, or 40 seconds
Everyone else: 5 lookups, or 150 seconds.
Cap/Sec team: 10 lookups, or 300 seconds.
Librarian: 20 lookups, or 90 seconds.  (But a lookup for him takes 4.5 seconds, as opposed to 30 seconds for everyone else.

> It sounds really interesting and I like that it gives roles to lesser positions, but generally modes that give crew lots of "voluntary" choices results in a complete shitshow.

Yes, and I took that into consideration.  I want to add a "law diploma" that cargo can order to give someone the ability to buy souls back like the lawyer/HoP.  This should alleviate the requirement for "voluntary" roundstart choices.

>  Also modes that involve conversion/snowballing/etc. tend to not go well with traitor at all.

lordpidey committed:
Fixes compile errors.
Re factors contracts to be subtypes rather than datums.  <--- incomplete, still need to summon them appropriately.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into lordpidey_demon

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into lordpidey_demon

lordpidey committed:
buffed infernal power contract to give passive healing
Nerfed magic power contract to not give robeless MM

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into lordpidey_demon

# Conflicts:
#	code/modules/mob/living/carbon/human/human.dm

lordpidey committed:
Merge branch 'master' into lordpidey_demon

# Conflicts:
#	_maps/map_files/MiniStation/MiniStation.dmm
#	code/datums/mind.dm

drynwyn commented:
If the demon dies, the soul-sellers should lose their powers.

lordpidey committed:
Fixes compile errors

lordpidey committed:
Corrects orientation of true demon sprite

Jazaen commented:
Do NOT let this die!

KorPhaeron commented:
>Do NOT let this die!

lordpidey committed:
Begins work on the sNPC for the true demon/ arch-demon

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	code/modules/hydroponics/hydroponics.dm
#	code/modules/spells/spell_types/ethereal_jaunt.dm

lordpidey committed:
Added funeral garment sprites.  These will be used in banishment rituals (hopefully)

lordpidey committed:
Stashing my work so I can move to another computer.

Re-working the true-demon, it will be a carbon rather than a simple mob.

lordpidey committed:
Revert "Stashing my work so I can move to another computer."

This reverts commit e8b1e032997b7e17af4ad8630bf21d3620195c4e.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into demon_local

Zilenan91 commented:
How's this going

PKPenguin321 commented:
well it compiles

lordpidey commented:
Current status:  Bashing my head against the wall trying to figure out how to make a new simple mob that does what I need it to.  I've got something close, but not 100% right.

KorPhaeron commented:
@lordpidey 

Would you like help?

lordpidey commented:
@KorPhaeron In about a week, if I can't figure something out, that would be helpful.

lordpidey committed:
Git exploded for me, so I have to make a new branch with EVERYTHING.  sorry.

lordpidey committed:
Merge branch 'demon_3' into lordpidey_demon

# Conflicts:
#	_maps/map_files/MetaStation/MetaStation.v41I.dmm
#	code/__DEFINES/contracts.dm
#	code/game/gamemodes/demon/demon_game_mode.dm
#	code/game/gamemodes/demon/demoninfo.dm
#	code/game/gamemodes/demon/true_demon/true_demon.dm
#	code/game/gamemodes/demon/true_demon/visuals_icons.dm
#	code/modules/mob/living/living_defines.dm
#	code/modules/spells/spell_types/demon.dm
#	icons/mob/32x64.dmi

lordpidey committed:
Fixes compile error

lordpidey committed:
Ghosts can now click on an arch demon to become a slaughter demon

lordpidey commented:
@KorPhaeron Aright, I've put some more effort into trying, and all I have are things that don't work. quite right.  If you could help, that would be appreciated.

What's needed at the moment, is a UI for the demon, as well as inhand sprite handling for the "true_demon" simple mob.

KorPhaeron commented:
@lordpidey 

Pop into IRC at some point then, will be easier to do it there than back and forth in github comments

lordpidey committed:
Begins to convert demons into a carbon rather than simple mob.

lordpidey committed:
Demons now resurrect if not banished.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	code/game/objects/items/weapons/defib.dm
#	code/modules/spells/spell_types/ethereal_jaunt.dm

lordpidey committed:
Beepsky now properly tells security chat about the level 666 threat.

lordpidey committed:
Contracts now work again.

lordpidey committed:
old spells now get removed properly.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

lordpidey committed:
adding pitchfork sprites

lordpidey committed:
Adds hud for the true-demon.  Not functional.

lordpidey committed:
Fixes another error in the demon hud.  Still broken.

lordpidey committed:
Demons are no longer immortal, demon huds work properly now.  (it's barebones, but it works.)

Jazaen commented:
Hm, ETA on testmerge readiness? Because having this added to april fools branch would be rad.

lordpidey commented:
@Jazen it's testmerged right NOW.

lordpidey committed:
Fixes the "Have mortals sign at least # contracts of TYPE" objective

lordpidey committed:
Fixed typo in banishlore() and updated wrath/envy sintouch objectives.

lordpidey committed:
Adds huds for demons/sintouched/soulless, however they are currently unused.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

lordpidey committed:
Updates the demon's hud to work with the recent hud changes.

lordpidey committed:
Cleaned up infernal jaunt, it works a lot better now and is less buggy.

lordpidey committed:
Revival contracts now actually take your soul.

lordpidey committed:
Fixes #16513

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	code/game/machinery/computer/arcade.dm
#	code/modules/mob/living/living.dm
#	tgstation.dme

lordpidey committed:
Replaces the infinite slaughter demons with infinite imps.  Sprites for imps are still needed.

lordpidey committed:
Adds sprite for imps.  It's a redder, smaller slaughter demon.

lordpidey committed:
Fixes lack of icon for pitchforks

lordpidey committed:
Gives summon wealth a more appropriate icon.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	icons/mob/inhands/items_righthand.dmi

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	code/modules/paperwork/photocopier.dm

lordpidey committed:
Fixes small part I forgot to merge.

Razharas commented:
Whats the progress on this?

lordpidey commented:
The only things left are a few bugs need fixing, and I'd like to have a wiki page done.  I have more plans after it gets merged for further improvements, such as making it an actual game-mode, rather than adminfuckery, but we can hold off on that for now.

https://tgstation13.org/wiki/User:Pidey/Sandbox is the wiki page so far, if you want to help improve it.

lordpidey committed:
Fixes a few bugs with demonic resurrection.  It's still very buggy.

lordpidey committed:
Derp, my bad, I didn't mean to admin myself on the main server.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	code/modules/mob/living/living.dm

lordpidey committed:
Fixes edge case of demonic resurrection failing.  Debraining the demon is NOT supposed to be an alternative for the banishment ritual.

lordpidey committed:
Also did not mean to change this config file.

lordpidey committed:
Fixes another error in type 1 demonic resurrection.  It works properly now.

lordpidey committed:
Updates employment contract text.

lordpidey committed:
Fixes type 2 resurrections.
Demonic contracts are once again permanently on fire.

lordpidey committed:
Replaces toy demonologist with toy demonomicon.  It still works the same, just different icon/name.

lordpidey committed:
Adds demonomicon and employment cabinet to box/meta/dream/efficiency.

I could not add it to mini/bird due to conflicts.

lordpidey committed:
Edits ministation and BirdStation to have demonomicons and employment cabinets.

lordpidey committed:
Fixes spelling error

lordpidey committed:
Adds burial garments to maps.

lordpidey commented:
holy CRAP birdboat station is a bitch to work with.  The map merger doesn't work, I had to open the file in notepad++ and figure out how to change it by hand.

KorPhaeron commented:
Just ignore birdboat for now

lordpidey commented:
Regardless, I've already done what I needed to in birdboat.  I'm not touching it again if I don't have to.

coiax commented:
You know that you're allowed to add yourself to the config/admins.txt because it's not the one we use on the main server? It makes it easier for testing shit on your own local server because then git won't complain at you constantly.

lordpidey commented:
Huh, I didn't realize that coiax.  Either way, I figured why change a file that doesn't need to be changed.

KorPhaeron commented:
@ChangelingRain @Jordie0608 @RemieRichards @Cheridan @duncathan 

Review please

lordpidey committed:
Update photocopier.dm

Fixes the span when inserting items into photocopier

AnturK commented on code/__DEFINES/hud.dm:
```diff
@@ -36,3 +36,6 @@
 #define ANTAG_HUD_NINJA 14
 #define ANTAG_HUD_CHANGELING 15
 #define ANTAG_HUD_ABDUCTOR 16
+#define ANTAG_HUD_DEMON	17
```
is this github viewer going wild or lack of space

lordpidey replied:
I was trying to keep all of the numbers aligned.  Look at the cult/rev/ops/wiz huds.

RemieRichards replied:
at that point you indent them all by 1 more.

AnturK commented on code/datums/mind.dm:
```diff
@@ -1616,6 +1672,27 @@
+/datum/mind/proc/disrupt_spells(var/delay, var/list/exceptions = New())
+	for(var/obj/effect/proc_holder/spell/S in spell_list)
+		var/found = 0
+		for(var/type in spells)
```
You probably meant in exceptions here, and you can just use continue instead of found and break.

lordpidey committed:
Fixes disrupt_spells proc

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	icons/obj/bureaucracy.dmi

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	_maps/map_files/MetaStation/MetaStation.v41I.dmm

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

Shadowlight213 commented:
@lordpidey Are demons considered human for the terms of asimov?

Zilenan91 commented:
I mean they're demons from the pits of hell so

Shadowlight213 commented:
But they have a human shell. Alsi they arent equipped to deal with borgs that well

Zilenan91 commented:
They can kill humans, so I assume they can kill borgs just as well since secborgs don't exist anymore.

PKPenguin321 commented:
it will probably be handled the same way as changelings, like how they're human until they do an explicitly non-human act

lordpidey commented:
I haven't made a firm decision, but I was leaning towards doing it like changlings.

Though I am open to suggestions.

coiax commented:
Well, once a devil has upgraded to blood lizard or above, the point is moot because it's all non-human from there.

Questions to answer

1. Is selling your soul harmful, or just probably bad but not actually against law one (akin to blowing up cloning)
2. Are humans who have sold their soul still human?
3. Are sintouched humans (the ones corrupted by the Demonomicon) still human?

My answers would be

1. No, but discouraged. And even if it is harmful, it's still willing self harm.
2. Yes.
3. Yes.

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+/proc/randomdemonbanish()
+	return pick(BANISH_WATER, BANISH_COFFIN, BANISH_FORMALDYHIDE, BANISH_RUNES, BANISH_CANDLES, BANISH_DESTRUCTION, BANISH_FUNERAL_GARB)
+
+/datum/demoninfo/proc/obligationlaw()
```
all these ```xlaw()``` and ```xlore()``` procs should be one ```/static``` list on ```/datum/demoinfo```
eg:
```
/datum/demoinfo
    var/list/laws_and_lore = list(
    OBLIGATION_FOOD = list("law" = "whatever the law text is", "lore" = "whatever the lore text is")
    OBLIGATION_DRINK = list("law" = "whatever the law text is", "lore" = "whatever the lore text is")
    ...)
```

lordpidey replied:
Some of the defines overlap though.  OBLIGATION_FOOD and BANE_SALT are both 1 for example.

Would it instead be okay to just make 8 static lists?

RemieRichards replied:
Just change those to not be numbers? those defines are only used (as far as I've seen) as organisational defines (they defeat typos, and give meaning to magic numbers), which means you can just change the values of those defines to be anything, even the exact same text like ```#define OBLIGATION_FOOD "obligation food"```

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+	if(SOULVALUE>0)
+		owner.current<< "<span class='userdanger'>Your body has been damaged to the point that you may no longer use it.  At the cost of some of your power, you will return to life soon.  Remain in your body.</span>"
+		sleep(DEMONRESURRECTTIME)
+		if (!body || (body && body.stat == DEAD))
```
the ```body``` part of ```body && body.stat == DEAD``` isn't necessary, as a null ```body``` will pass ```if (!body```

RemieRichards commented on code/game/gamemodes/demon/imp/imp.dm:
```diff
@@ -0,0 +1,57 @@
+							Though you are not obligated to help, perhaps by aiding a higher ranking demon, you might just get a promotion.  However, you are incapable	\
+							of intentionally harming a fellow demon.</B>"
+
+/mob/living/simple_animal/slaughter/New()
```
slaughter?

isn't this the Imp file?

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+			if(SOULVALUE>=POWERUPTHRESHOLD*2) //Yes, BOTH this and the above if statement are to run if soulpower is high enough.
+				var/mob/living/carbon/true_demon/A = new /mob/living/carbon/true_demon(targetturf)
+				A.faction |= "hell"
+				H.loc = A
```
all ```thing.loc = X``` should be ```thing.forceMove(X)```

lordpidey replied:
even if X isn't a turf?

RemieRichards replied:
that's correct

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+				H.socks = "Nude"
+				H.dna.features["mcolor"] = "511"
+				H.regenerate_icons()
+			if(SOULVALUE>=POWERUPTHRESHOLD*2) //Yes, BOTH this and the above if statement are to run if soulpower is high enough.
```
spacing on these operators and operands please.

RemieRichards commented on code/game/gamemodes/demon/objectives.dm:
```diff
@@ -0,0 +1,63 @@
+
+/datum/objective/demon/soulquantity/check_completion()
+	var/count = 0
+	for(var/datum/mind/L in owner.demoninfo.soulsOwned)
```
this should be an istypeless forloop, provided ```soulsOwned``` is a list of just ```/datum/mind```s

RemieRichards commented on code/game/gamemodes/demon/objectives.dm:
```diff
@@ -0,0 +1,63 @@
+	for(var/datum/mind/L in owner.demoninfo.soulsOwned)
+		if(L.soulOwner != L)
+			count++
+	return count>=quantity
```
operator spacing

RemieRichards commented on code/game/gamemodes/demon/objectives.dm:
```diff
@@ -0,0 +1,63 @@
+
+/datum/objective/demon/soulquality/check_completion()
+	var/count = 0
+	for(var/datum/mind/L in owner.demoninfo.soulsOwned)
```
this should be an istypeless forloop, provided ```soulsOwned``` is a list of just ```/datum/mind```s

RemieRichards commented on code/game/gamemodes/demon/true_demon/_true_demon.dm:
```diff
@@ -0,0 +1,159 @@
+		if(has_bane(BANE_LIGHT))
+			mind.disrupt_spells(-500)
+			return ..() //flashes don't stop demons UNLESS it's their bane.
+	return
```
trailing return, remove it please.

RemieRichards commented on code/game/gamemodes/demon/true_demon/inventory.dm:
```diff
@@ -0,0 +1,94 @@
+/mob/living/carbon/true_demon/activate_hand(selhand)
```
this entire proc is inherited from ```/carbon```, with no edits.

RemieRichards commented on code/game/gamemodes/demon/true_demon/inventory.dm:
```diff
@@ -0,0 +1,94 @@
+		mode()
+
+
+/mob/living/carbon/true_demon/swap_hand()
```
this entire proc is inherited from ```/carbon```, with no (functional) edits.

RemieRichards commented on code/game/machinery/dna_scanner.dm:
```diff
@@ -108,7 +108,7 @@
 			|| locate(/obj/machinery/computer/cloning, get_step(src, EAST)) \
 			|| locate(/obj/machinery/computer/cloning, get_step(src, WEST)))
-			if(!occupant.suiciding && !(occupant.disabilities & NOCLONE))
+			if(!occupant.suiciding && !(occupant.disabilities & NOCLONE) && !occupant.hellbound)
```
we really need some kind of ```can_revive()```/```can_clone()``` procs, this isn't your fault or responsibility though.

RemieRichards commented on code/game/objects/items/weapons/twohanded.dm:
```diff
@@ -443,3 +443,66 @@
+
+/obj/item/weapon/twohanded/pitchfork/update_icon()  //Currently only here to fuck with the on-mob icons.
+	icon_state = "pitchfork[wielded]"
+	return
```
trailing return, remove it please.

RemieRichards commented on code/game/objects/items/weapons/twohanded.dm:
```diff
@@ -443,3 +443,66 @@
+				H.apply_damage(rand(force/2, force), BURN, pick("l_arm", "r_arm"))
+			else
+				U.adjustFireLoss(rand(force/2,force))
+		return
```
trailing return, remove it please.

RemieRichards commented on code/modules/library/lib_demonomicon.dm:
```diff
@@ -0,0 +1,56 @@
+			if(!prob(correctness))
+				usedName += "x"
+			var/datum/demoninfo/demon = demonInfo(usedName, 0)
+			user << browse("Information on [demonName]<br><br><br>[demon.banlore()]<br>[demon.banelore()]<br>[demon.obligationlore()]<br>[demon.banishlore()]", "window=book[window_size != null ? ";size=[window_size]" : ""]")
```
this is a very old form of UI, you should atleast look into the ```/datum/browser``` UIs, if not a proper tgui interface (like, actually making it look like a demonimocon)

lordpidey replied:
I'll probably need some help with this one.  If only because my artistic skills are so bad, that stick men sue me for defamation when I draw.

RemieRichards replied:
Oh no, I'm not saying this /requires/ a fancy tgui interface with cool graphics, I'm saying you should atleast look into a ```/datum/browser``` UI (look at procs like Interact() and the browser datum's file) and if you have time later/some help look into a tgui one.

RemieRichards commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -960,5 +961,43 @@
+	src.mind.objectives += O
+	var/obj_count = 1
+	src << "<span class='notice'>Your current objectives:</span>"
+	for(var/datum/objective/objective in src.mind.objectives)
```
this should be an istypeless forloop.

RemieRichards commented on code/modules/mob/new_player/new_player.dm:
```diff
@@ -342,6 +343,14 @@
+/mob/new_player/proc/AddEmploymentContract(var/mob/living/carbon/human/employee)
+	//TODO:  figure out a way to exclude wizards/nukeops/demons from this.
+	sleep(30)
+	for(var/obj/structure/filingcabinet/employment/employmentCabinet in employmentCabinets)
```
this should be an istypeless forloop.

RemieRichards commented on code/modules/paperwork/contract.dm:
```diff
@@ -0,0 +1,249 @@
+
+/obj/item/weapon/paper/contract/infernal/knowledge/FulfillContract(mob/living/carbon/human/user = target.current)
+	user.dna.add_mutation(XRAY)
+	for(var/datum/atom_hud/H in huds)
```
this should be an istypeless forloop.

RemieRichards commented on code/modules/paperwork/filingcabinet.dm:
```diff
@@ -165,3 +166,59 @@
+
+/obj/structure/filingcabinet/employment/proc/fillCurrent()
+	//This proc fills the cabinet with the current crew.
+	for(var/datum/data/record/G in data_core.locked)
```
this should be an istypeless forloop.

RemieRichards commented on code/modules/reagents/chemistry/reagents/food_reagents.dm:
```diff
@@ -236,6 +236,12 @@
 	color = "#FFFFFF" // rgb: 255,255,255
 
+/datum/reagent/consumable/sodiumchloride/reaction_mob(mob/living/M, method=TOUCH, reac_volume)//Splashing people with welding fuel to make them easy to ignite!
+	if(!istype(M, /mob/living))
```
just ```if(!istype(M))``` would work

RemieRichards commented on code/modules/reagents/reagent_containers.dm:
```diff
@@ -38,6 +38,8 @@
 /obj/item/weapon/reagent_containers/attack(mob/M, mob/user, def_zone)
+	if(user.a_intent == "harm")
+		return ..()
 	return
```
trailing return, remove it please.

RemieRichards commented on code/modules/spells/spell_types/demon.dm:
```diff
@@ -0,0 +1,166 @@
+
+	proj_icon_state = "fireball"
+	proj_name = "a fireball"
+	proj_type = "/obj/effect/proc_holder/spell/turf/fireball/demon"
```
>string typepath

lord no.

RemieRichards commented on code/game/gamemodes/demon/demon_game_mode.dm:
```diff
@@ -0,0 +1,103 @@
+	if(sintouched.len)
+		text += "<br><span class='big'><b>The sintouched were:</b></span>"
+		var/list/sintouchedUnique = uniqueList(sintouched)
+		for(var/datum/mind/sintouched_mind in sintouchedUnique)
```
this should be an istypeless forloop.

RemieRichards commented on code/game/gamemodes/demon/demon_game_mode.dm:
```diff
@@ -0,0 +1,103 @@
+	/var/text = ""
+	if(demons.len)
+		text += "<br><span class='big'><b>The demons were:</b></span>"
+		for(var/datum/mind/demon in demons)
```
this should be an istypeless forloop.

RemieRichards commented on code/game/gamemodes/demon/demon_game_mode.dm:
```diff
@@ -0,0 +1,103 @@
+	current << "<br/><br/> <span class='warning'>Remember, the crew can research your weaknesses if they find out your demon name.</span><br>"
+	var/obj_count = 1
+	current << "<span class='notice'>Your current objectives:</span>"
+	for(var/datum/objective/objective in objectives)
```
this should be an istypeless forloop.

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+
+#define DEMONRESURRECTTIME 600
+
+var/list/allDemons = list()
```
```var/global/list/allDemons = list()```

RemieRichards commented on code/datums/mind.dm:
```diff
@@ -1616,6 +1672,24 @@
 		var/obj/effect/proc_holder/spell/S = X
 		S.action.Grant(new_character)
 
+/datum/mind/proc/disrupt_spells(var/delay, var/list/exceptions = New())
```
no ```var/``` in proc arguments

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+	var/reviveNumber = 0
+	var/form = BASIC_DEMON
+
+/proc/randomDemonInfo(var/name = randomDemonName())
```
no ```var/``` in proc arguments

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+	demon.banish = randomdemonbanish()
+	return demon
+
+/proc/demonInfo(var/name, var/saveDetails = 1)
```
no ```var/``` in proc arguments

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+		if(BANISH_FUNERAL_GARB)
+			return "Funeral garments will prevent the demon from resurrecting."
+
+/datum/demoninfo/proc/add_soul(var/datum/mind/soul)
```
no ```var/``` in proc arguments

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+		if(POWERUPTHRESHOLD*3)
+			increase_arch_demon()
+
+/datum/demoninfo/proc/remove_soul(var/datum/mind/soul)
```
no ```var/``` in proc arguments

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+	owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/summon_contract(null))
+
+
+/datum/demoninfo/proc/give_base_spells(var/give_summon_contract = 0)
```
no ```var/``` in proc arguments

RemieRichards commented on code/modules/mob/new_player/new_player.dm:
```diff
@@ -342,6 +343,14 @@
 					var/obj/machinery/announcement_system/announcer = pick(announcement_systems)
 					announcer.announce("ARRIVAL", character.real_name, rank, list()) //make the list empty to make it announce it in common
 
+/mob/new_player/proc/AddEmploymentContract(var/mob/living/carbon/human/employee)
```
no ```var/``` in proc arguments

coiax commented:
oh shit, Joan just reviewed the crap out of your code

RemieRichards commented:
>crediting Joan with my work.

coiax commented:
> > crediting Joan with my work

Nope, I'm pretty sure it says Remie

Shadowlight213 commented:
Remie and Joan confirmed for same person.
It's all a conspiracy so they can selfmerge.

RemieRichards commented:
@coiax 
![capture](https://cloud.githubusercontent.com/assets/4043781/14530736/2d8d87ea-0252-11e6-957b-344b0d2a5a95.PNG)

you lose.

coiax commented:
@RemieRichards 

But my email clearly says this
![obviously photoshopped email screenshot](http://i.imgur.com/wxaHxDt.png)

RemieRichards commented:
For a brief moment I was amazed how you'd managed to snip the image so similarly to mine.
before I realised it IS my image you edited, rofl.

Shadowlight213 commented:
@lordpidey Oh, also is it possible to somehow make the demon immune to the incinerator? Because I saw the crew literally not try to research the demon, but instead shoved them into the incinerator before they could revive similar to lings.(I think they thought it was a ling actually when they saw it self revive)
Kinda makes the whole research the name to prevent revives mechanic pointless.

lordpidey commented:
@RemieRichards Thank you for the review, I should be able to finish it later today.

On a related note, why are typeless for loops better than typed ones?  Isn't the extra sanity check helpful?

@Shadowlight213 If the demon's body is destroyed, the demon will respawn in a new body (Provided that body destruction isn't the banishment ritual).  However, that was bugged in the March30/31st test run, and didn't work.

Shadowlight213 commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,474 @@
+	give_lizard_spells()
+	form = BLOOD_LIZARD
+
+
```
O

lordpidey commented:
> Well, once a devil has upgraded to blood lizard or above, the point is moot because it's all non-human from there.
>
>Questions to answer
>
>    Is selling your soul harmful, or just probably bad but not actually against law one (akin to blowing up cloning)
>    Are humans who have sold their soul still human?
>    Are sintouched humans (the ones corrupted by the Demonomicon) still human?
>
>My answers would be
>
>    No, but discouraged. And even if it is harmful, it's still willing self harm.
>    Yes.
>    Yes.

I agree on all three of those, with the exception of the revival contract, as it heals/revives you, **sets you on fire**, waits two seconds, extinguishes you, and then heals all burn damage.

RemieRichards commented:
@lordpidey I only pointed it out for homogeneous lists, ones who's data is of the same type or subtypes of the type being accessed, in these cases the istypes are completely unnecessary and add some overhead.

Shadowlight213 commented:
Oh. Before I forget. We did a test run yesterday, and I noticed the demon didn't show up on the check antag list, nor did it show up on the antag hud.

coiax commented:
If the knowledge boon from selling your soul is still giving a person antagHUD, it's probably a good time to check that antagHUD is working correctly.

lordpidey commented:
@Shadowlight213 I'm not very familiar with admintools, where is the check antag list?

lordpidey commented:
@RemieRichards so it's faster to use a typeless for loop, then cast the var?  I thought the casting added more overhead than the typing.  I'll change that, and keep that in mind in the future.

RemieRichards commented:
/only/ for homogenous lists @lordpidey , think like the "silicon mobs" list or the "airlock" list or the "minds" list, if the list has MIXED data, you MUST do a typed loop, for safety.

Shadowlight213 commented:
@lordpidey
https://github.com/tgstation/-tg-station/blob/master/code/modules/admin/player_panel.dm#L311

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	_maps/map_files/TgStation/tgstation.2.1.3.dmm
#	code/game/machinery/computer/arcade.dm
#	code/game/objects/items/toys.dm

Shadowlight213 commented:
@lordpidey merge conflicts

Shadowlight213 commented:
@lordpidey OK. MAJOR ISSUE. TRUE DEMONS DISAPPEAR FROM THE PLAYER PANEL AND ADMINS CAN'T PM THEM AHHHH.

Shadowlight213 commented:
Also true demons have no tongues

Shadowlight213 commented:
Another bug report from a player.
Apparently if you resist as an archdemon, while on fire, you get stuck and cant get up or something.

lordpidey committed:
Makes a lot of changes as reccomended by Remie.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	code/modules/mob/living/living.dm

lordpidey committed:
Fixes a compile error.

coiax commented:
they're DEVILS not DEMONS REEEEEEEEEEEEEEEEEEEEEEEEEEEEEEeeee

lordpidey committed:
Updates the lawify/loreify to be lists.  Does not compile.

lordpidey committed:
Fixes compile errors.

lordpidey committed:
Arch demons no longer regress upon death.
Speaking a demon's truename gives the demon the opportunity to teleport to you.
Makes demonomicon not care about capitalization.

lordpidey committed:
Fixes startup error.  Demonic summoning now works.  It's very simple, just say their true name.

lordpidey committed:
Demons now have a tongue.  True demon bodies are deleted upon regression.

lordpidey committed:
Demons can now be punched.
Demons can no longer resist a fire out, since it does nothing to them.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	code/game/machinery/computer/arcade.dm

lordpidey committed:
true demons show up in player panel.  Hellfire works again.

lordpidey commented:
Aright.  I've fixed all bugs I know of.

coiax commented:
THEY'RE DEVILS REEEEEEEEEEEEEEEEEEEEEEE LAWFUL EVIL REEEEEEEEEEEEEE

duncathan commented:
coiax is right

lordpidey commented:
Aright, aright, I'll post a poll on the forums in the morning about what to name them.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	code/__DEFINES/tick.dm

lordpidey commented:
https://tgstation13.org/phpBB/viewtopic.php?f=9&t=6443

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	code/modules/mob/living/living.dm
#	icons/mob/inhands/items_lefthand.dmi
#	icons/mob/inhands/items_righthand.dmi

lordpidey committed:
Another day, another merge conflict fixed.

# Conflicts:
#	icons/mob/actions.dmi

lordpidey committed:
Prepares for commit 16940

KazeEspada commented:
Heavily biased towards naming them devils. Mosly because of D&D but also because it makes them named different than slaughter demons.

";DEVIL IN THE LAWYER OFFICE!"

lordpidey committed:
Demons are now known as devils.

Still needs testing.

lordpidey committed:
Oops, didn't commit everything last commit.

lordpidey committed:
Finishes converting demons to devils.

RemieRichards commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -255,8 +255,23 @@
+	speakDemonName(message)
 	return message
 
+/obj/item/organ/internal/tongue/proc/speakDemonName(var/message)
```
oh lord no.

RemieRichards commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -257,8 +257,20 @@
+		speakDevilName(message)
 	return message
 
+/obj/item/organ/internal/tongue/proc/speakDevilName(var/message)
```
no no no no no no no.

lordpidey replied:
Care to explain why?  Unless there is a devil, this proc will simply see that the list is empty and do nothing.  Even if there's three devils, it will only be about as intensive as the lizardman speech replacement.

RemieRichards replied:
It's incredibly snowflakey

lordpidey replied:
How or where would you recommend implementing this so that it's not snowflakey?

RemieRichards replied:
I see no alternative, I simply suggest not doing this feature.

coiax replied:
How about implementing speech activated spells, then giving every living mob the SUMMON THE DEMON spell secretly. Still means you have speech scanning, but in a more general form.

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,397 @@
+#define DEVILRESURRECTTIME 600
+
+var/global/list/allDevils = list()
+var/static/list/lawlorify = list (
```
since you've made this properly global, define it as ```/global``` not ```/static```

lordpidey replied:
Wouldn't static be more appropriate, as the list will never change?

RemieRichards replied:
static and global are synonymous, but are used differently for organisational purposes, when I suggested putting it on the TYPE, static was appropriate, now it is off the type global is more appropriate.

lordpidey replied:
Okay, I'll change it, thought I still don't understand the difference between when to use static and global.

lordpidey committed:
Fixes merge conflicts.

lordpidey committed:
Fixes speech for lizards/flymen when they speak a demon's name.

KazeEspada commented on config/tips.txt:
```diff
@@ -176,3 +176,5 @@ As a Morph, you can talk while disguised, but your words have a chance of being
 As a Ghost, you can see the inside of a container on the ground by clicking on it.
 As a Ghost, you can double click on people, bots, or the singularity to follow them.
+As a Demon, you gain power for every three souls you control, however you also become more obvious.
+As a Demon, as long as you control at least one other soul, you will automatically resurrect, as long as a banishment ritual is not performed.
```
Devil

KazeEspada commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -290,6 +304,8 @@
 
 /obj/item/organ/internal/tongue/fly/TongueSpeech(var/message)
+	spawn(0)
+		speakDevilName(message)
```
A proc that runs everytime you speak, just to make sure you didn't utter a truename?

lordpidey replied:
Yes, neither I nor Remie could see a better way of implementing this.  However, I am of the belief that while somewhat snowflakey, this is a feature that will add a lot to the antag.

I will look for a better way to implement this, but tongue seemed like a very good place, as I would need the message pre speech impediment, and also to prevent unneeded checks for when xenos/drones/sillicons speak.

AnturK replied:
Implement simple event system. Hook it to that. Also helps with organization of  the whole on death effects and on other reactionary effects. I'll understand if you don't want to do that though since it would be a lot of work.

lordpidey committed:
Update tips.txt

lordpidey committed:
Changes variable from static to global as per Remie's suggestion.

I disagree with Remie's assessment, as a list that will never change, even between games, seems like it should be static.

torger597 commented:
:+1: 
Test merge ended before I could try it out but it seems absolutely amazing

coiax commented:
An extensive discussion in #coderbus suggested that  you remove the truename detection from all speech, and just put hooks in

- lit flares (includes lit torches)
- lit welding torches
- candles

To detect the name being spoken nearby.

lordpidey commented:
I'll agree that the tongue was a poor place to have the check.

And I like your hooks in candles/flaming things idea,

But I was already working on another method, I just needed to test it.

Anyway:  take a look at https://github.com/lordpidey/-tg-station/commit/9b610f6febe21541163997235fa6a820c565af73

Is that an acceptable way of implementing it?

coiax commented:
To my eyes it certainly looks better, but I think most of the coder's objection is putting hooks directly into every part of speech.

I like the no-shouting-in-the-library idea though. If this doesn't pass merit, I suggest that you make a seperate PR, applying some sort of NOSHOUTING debuff to anyone who enters the /area/library zone.

KorPhaeron commented:
Antag hud is still probably a bad idea. It will definitely be abused

Zilenan91 commented:
As is the people keeping their powers if they get slapped by the lawyer.

People are just gonna accept a contract, run to the lawyer/HoP, and then get free powers at no cost.

lordpidey committed:
Removes devil summoning.  Hopefully, I'll be able to eventually find a way to implement it satisfactorily, but for now, I'll leave it out.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	icons/mob/actions.dmi

lordpidey commented:
@KorPhaeron I'm not quite sure about that.  The boons have to be good enough to warrant bringing the station closer to having an arch demon on it.

Even ordinary blood lizards are quite robust.

@Zilenan91, that's intentional.  The demon WILL have to do something about the lawyer, or the contracts.

coiax commented:
Having realise how snowlakey Pierre's Throat code is, I feel it's unfair to deny devils the ability to be namesummoned when there's a check running on all speech code for being under the effects of a rare disease. Fire summing is still cool.

And antagHUD will not stay secret for long, it'll give you a target on your head from every single antag. If we want a suggestion, can we drop the knowledge boon to just antagHUD (because xray is hard to fake), and make it fakeable in some way? Because then security will have to trust that the tradee is telling the truth, and isn't just trying to throw them off.

RemieRichards commented:
@coiax No, that is literally the argument of "This section of code is shit, so lets make it worse", Bad logic, Not happening.

coiax commented:
Why don't we just take his HandleSpeech() idea of a proc, move the pierre's throat to there, and have anything that's added to the list of checks be very quick in determining whether to spend any amount of time on it. Pierre's throat is a disease presence/absence, truename summoning is demons being in the round.

lordpidey commented:
@coiax  That sounds like a good idea.... but my reserves of effort is running a bit low at the moment.  Perhaps we should implement it without summoning, and then later add it if it seems appropriate.

coiax commented:
@lordpidey you've already done the work with making a system for hooks, I'll just see if I can add pierre's throat to that list of special speech checking and make a PR for you tommorow morning.

lordpidey commented:
Neither of the hooks I made support changing the message though.  One is simply for blocking the WHOLE message, the other is for effects that happen upon speaking.

coiax commented:
> tommorow morning

I underestimate how long things take to do.

coiax committed:
Removes carriage returns

coiax committed:
Merge branch 'master' into merged_demon

Conflicts:
	icons/mob/actions.dmi
	icons/mob/mob.dmi
	tgstation.dme

coiax committed:
Combined modified icon files

coiax committed:
Merge branch 'master' into merged_demon

Conflicts:
	code/modules/mob/living/carbon/human/species.dm

coiax commented:
I have made a PR to merge all of the stuff from master into this branch at https://github.com/lordpidey/-tg-station/pull/1

We need to get this merged into master soon as possible otherwise these merge conflicts are going to KEEP COMING. It's stable enough, admin spawn only, and having enough hooks in the code in mainline mean it'll be harder to break something.

lordpidey committed:
Merge pull request #2 from coiax/merged_demon

Merged demon

lordpidey committed:
Fixes #17184

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

Shadowlight213 commented:
@lordpidey the true and arch devil sprites have their icon states still refer to demon, yet are set to devil, which makes their icons break.

lordpidey committed:
Imps no longer show up in the end round report.  This is because there can EASILY be 100+ imps if the crew is well armed.

Non-employment filing cabinets no longer take 30 sec to wrench/unwrench.

lordpidey committed:
The lawyer can actually buy souls back now.  Oops.

lordpidey commented:
@shadowlight213 The imp slowdown is intentional, but if you think it's best without, then I can remove it.

I'll take a look at the sprite issue.

The only reason I can think of for prestige failing is a drone signing it.  If all else fails, the card should simply drop to the feet of the signer.

lordpidey committed:
Fixes true/arch devil spritesheet to have correct sprite names.

Razharas commented:
How ready is this?

lordpidey commented:
I just need to make one more mapedit, and then it's ready.  Expect a push or two in the next hour.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

lordpidey committed:
Relocates Box's employment cabinet to Law office.

lordpidey committed:
De-devil-ing someone now removes the devilinfo.
De-devil-ing now returns an error when used on true and arch devils

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into lordpidey_demon

# Conflicts:
#	code/modules/mob/living/living.dm

Core0verload commented on code/game/gamemodes/demon/true_demon/_true_demon.dm:
```diff
@@ -0,0 +1,191 @@
+#define DEVIL_TOTAL_LAYERS 2
+
+
+/mob/living/carbon/true_devil
```
Adding another carbon? Always a bad idea.

lordpidey replied:
Why?  It needs to be able to hold things, and needs reagent handling.

Core0verload replied:
Make it human species then.

lordpidey replied:
It doesn't have any other human slots though.

Core0verload replied:
You can ban him from other human slots in species datum. Or make him only able to wear specific items in these slots. That's easy. Easier than managing yet another ultra special carbon.

Razharas replied:
I agree, species should in general replace carbons

Core0verload commented on code/modules/clothing/under/miscellaneous.dm:
```diff
@@ -243,6 +243,15 @@
+	icon_state = "burial"
+	item_state = "burial"
+	item_color = "burial"
+	whiteness = 1
```
I don't like that you are adding special vars just for devil side antag. Make it a list path = value somewhere in devil's code.

lordpidey committed:
Re-factors whiteness code for jumpsuits.

KorPhaeron commented:
@lordpidey 

What if knowledge increased your client view range to 10 or so as well as giving xray? That'd be a pretty significant (and unique) advantage and it would solve my biggest issue with this PR (antag hud)

lordpidey commented:
@KorPhaeron

I'm okay with that idea on a balance level, however, I am concerned over bandwidth usage.  Wouldn't that increase bandwidth usage by about 50%?

I don't have access to the server's bandwidth logs, so I don't know how much of a problem that would be, so if you say it won't be a problem, then I'll take your word on it.

KorPhaeron commented:
Actually I have no idea! All I know is it's pretty nice to extend your view range while observing as an admin. Only a handful of people would have this, right?

@MrStonedOne 

Would it impact things too much if some players got to see 50% further?

coiax commented:
@KorPhaeron that is a brilliant idea. 10/10.

The only problem with that is that your HUD drops to the bottom of that view range, but that shouldn't be a problem to anyone with a large screen. Possibly enable picking a smaller view range than 10 in that case?

coiax committed:
Merge branch 'master' into devil-3

Conflicts:
	_maps/map_files/TgStation/tgstation.2.1.3.dmm
	icons/mob/inhands/items_lefthand.dmi
	icons/mob/inhands/items_righthand.dmi
	icons/mob/mob.dmi
	icons/obj/bureaucracy.dmi

coiax committed:
Merges icons, reverts failed merge of map

coiax committed:
Merges map with CAS decks

lordpidey committed:
Merge pull request #4 from coiax/devil-3

Fixes merge conflicts

coiax committed:
Merge remote-tracking branch 'origin/master' into devil-5

lordpidey committed:
Merge pull request #5 from coiax/devil-5

Devil 5

coiax committed:
Merge branch 'master' into devil-6

MrStonedOne commented:
With the way byond does everything, it wouldn't lead to too much extra usage.

lordpidey committed:
Merge pull request #6 from coiax/devil-6

Devil 6

coiax committed:
Merge branch 'master' into devil-7

Shadowlight213 commented:
I'm fairly certain vg allows ghosts to zoom in and out as they please, so it doesn't sound like that would be a performance issue.

lordpidey committed:
Merge pull request #7 from coiax/devil-7

Devil 7 (otherwise known as: OTHER PEOPLE, PLEASE STOP EDITING mob.dmi)

lordpidey commented:
Okay, tomorrow after my last final, I'll change the antag hud to a verb that lets you change vision range.

coiax commented:
Admins already have a verb for that, just put a cap on the range so it can't be higher than 10, and also make it an action button.

lordpidey commented:
Yeah, that's what I was planning on doing.

Dorsisdwarf commented:
I wouldn't make it 10 for living mobs, there's noticable slowdown at 10, I seem to recall

lordpidey committed:
Merge branch 'master' into lordpidey_demon

# Conflicts:
#	icons/mob/actions.dmi
#	icons/mob/hud.dmi

lordpidey committed:
Fixes copying employment contracts

lordpidey committed:
Derp, fixes compile error.

lordpidey committed:
Replaces antaghud with customizable vision range for knowledge boon.

lordpidey committed:
Merge remote-tracking branch 'origin/master' into lordpidey_demon

# Conflicts:
#	_maps/map_files/TgStation/tgstation.2.1.3.dmm
#	code/modules/mob/living/carbon/human/species.dm

KorPhaeron commented:
It's admin only, if the code is okay I don't think merging it for longer term testing is a bad thing

KorPhaeron commented:
@RemieRichards @ChangelingRain @ChangelingRain @phil235 @AnturK

phil235 commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,398 @@
+	var/mob/living/carbon/true_devil/D = owner.current
+	D << "<span class='warning'>You feel as though your form is about to ascend."
+	sleep(50)
+	D.visible_message("<span class='warning'>[D]'s skin begins to erupt with spikes.</span>", \
```
if the mob gets gibbed during the sleep() you get a runtime.

lordpidey committed:
Prevents a runtime if devil is gibbed while ascending.

lordpidey committed:
Merge branch 'master' into lordpidey_demon

lordpidey committed:
Merge remote-tracking branch 'origin/master' into lordpidey_demon

# Conflicts:
#	icons/mob/hud.dmi

KorPhaeron commented:
This is gonna conflict forever, I suggest finding a maintainer willing to merge it before you fix conflicts again

lordpidey committed:
Merge branch 'master' into lordpidey_demon

# Conflicts:
#	code/game/machinery/computer/arcade.dm
#	icons/mob/mob.dmi

AnturK commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -3,7 +3,8 @@
 	icon = 'icons/mob/human.dmi'
-	icon_state = "caucasian1_m_s"
+	icon_state = "caucasian1_m_s"
+	var/acedia = 0
```
Add a function that checks for objective type in the objective list instead of adding this please.

AnturK commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -988,5 +989,44 @@
+		src << "<B>Objective #[obj_count]</B>: [objective.explanation_text]"
+		obj_count++
+
+/mob/living/proc/check_weakness(obj/item/weapon, mob/living/attacker)
```
Wouldn't mind moving this check down to species.

AnturK commented on code/modules/mob/new_player/new_player.dm:
```diff
@@ -342,6 +343,15 @@
 
+/mob/new_player/proc/AddEmploymentContract(mob/living/carbon/human/employee)
+	//TODO:  figure out a way to exclude wizards/nukeops/demons from this.
+	sleep(30)
```
Why is this sleep here ?

lordpidey replied:
To give Ragin Mages a chance to work right.  I'm not sure if it's still needed, I'll look into it.

RemieRichards replied:
did you look into this?

AnturK commented on code/modules/paperwork/contract.dm:
```diff
@@ -0,0 +1,265 @@
+	info = "<center>Conditions of Employment</center><BR><BR><BR><BR>This Agreement is made and entered into as of the date of last signature below, by and between [target] (hereafter referred to as SLAVE), and Nanotrasen (hereafter referred to as the omnipresent and helpful watcher of humanity).<BR>WITNESSETH:<BR>WHEREAS, SLAVE is a natural born human or humanoid, posessing skills upon which he can ...
+
+
+/obj/item/weapon/paper/contract/employment/attack(mob/living/M, mob/living/carbon/human/user)
```
Adding NOBLUDGEON flag to the contract and adding ..() to attack procs would be better here.

lordpidey replied:
Done.  Not sure why this comment isn't hidden now.

RemieRichards replied:
because AnturK commented on a line you haven't changed, so the diff is the same and as such the comment remained.

lordpidey committed:
Neatens contract code, and removes excess variable from humans.

Shadowlight213 commented:
@lordpidey failing travis.

lordpidey committed:
Fixes compile errors

lordpidey commented:
Right, sorry about that.  Should be fixed now.  I'm still working on moving check_weakness down to species.

lordpidey committed:
Organizes weakness code slightly.  Adds ability for a species override on weakness.  (Make flypeople take 2x damage from flyswatters for example)

coiax commented:
@lordpidey just get it merged, we can improve it later

lordpidey commented:
Nice timing there @coiax, you managed to say that less than a MINUTE after I comitted it.

Shadowlight213 commented:
@lordpidey so should I start mergebegging it? Or do you still need to work on it?

lordpidey commented:
Yes, let us begin the mergebegging.

lordpidey committed:
Removes sixteen erroneous characters from the codebase.

RemieRichards commented on code/game/gamemodes/demon/imp/imp.dm:
```diff
@@ -0,0 +1,57 @@
+	visible_message("<span class='danger'>[src] screams in agony as it sublimates into a sulfurous smoke.</span>")
+	ghostize()
+	qdel(src)
+	return
```
trailing return, remove it please.

lordpidey committed:
Removes trailing return

RemieRichards commented on code/modules/admin/player_panel.dm:
```diff
@@ -592,6 +592,33 @@
 
+		if(ticker.mode.devils.len)
+			dat += "<br><table cellspacing=5><tr><td><B>devils</B></td><td></td><td></td></tr>"
+			for(var/datum/mind/devil in ticker.mode.devils)
```
this should be an istypeless forloop.

RemieRichards commented on code/modules/admin/player_panel.dm:
```diff
@@ -592,6 +592,33 @@
+
+		if(ticker.mode.sintouched.len)
+			dat += "<br><table cellspacing=5><tr><td><B>sintouched</B></td><td></td><td></td></tr>"
+			for(var/datum/mind/sintouched in ticker.mode.sintouched)
```
this should be an istypeless forloop.

RemieRichards commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -990,3 +990,41 @@
+	var/obj_count = 1
+	src << "<span class='notice'>Your current objectives:</span>"
+	for(O in src.mind.objectives)
+		var/datum/objective/objective
```
this isn't ever set to O, so ```objective.explanation_text``` below will runtime.

RemieRichards commented on code/modules/mob/living/living.dm:
```diff
@@ -995,3 +995,29 @@ Sorry Giacom. Please don't be mad :(
+	if(mind)
+		mind.soulOwner = mind
+
+/mob/living/proc/has_bane(var/banetype)
```
no ```var/``` in proc arguments

RemieRichards commented on code/modules/paperwork/contract.dm:
```diff
@@ -0,0 +1,270 @@
+	if(!istype(user) || !user.mind)
+		return -1
+	user.dna.add_mutation(XRAY)
+	//Apparently antag-hud is too powerful.
```
Don't comment out unwanted code, remove it.

lordpidey committed:
Makes typeless for loops, removes a runtime, and removes unhelpful comments.

RemieRichards commented:
I'm... I'm gonna do it!

Jazaen commented:
\o/

RemieRichards commented:
>132 commits

thank god I squashed it.

PKPenguin321 commented:
:Tada:

phil235 commented on code/game/gamemodes/demon/demon_game_mode.dm:
```diff
@@ -0,0 +1,108 @@
+/datum/game_mode/proc/update_sintouch_icons_removed(datum/mind/sin_mind)
+	var/datum/atom_hud/antag/hud = huds[ANTAG_HUD_SINTOUCHED]
+	hud.leave_hud(sin_mind.current)
+	set_antag_hud(sin_mind.current, null)
```
these two procs do nothing, they are never called.
